subsys/bluetooth/host/conn.c

```diff
@@ -118,7 +118,9 @@ sys_slist_t bt_auth_info_cbs = SYS_SLIST_STATIC_INIT(&bt_auth_info_cbs);
 #endif /* CONFIG_BT_SMP || CONFIG_BT_CLASSIC */
 
 
+#if defined(CONFIG_BT_CONN_DYNAMIC_CALLBACKS)
 static sys_slist_t conn_cbs = SYS_SLIST_STATIC_INIT(&conn_cbs);
+#endif /* CONFIG_BT_CONN_DYNAMIC_CALLBACKS */
```

Comment:
Correct me if I'm wrong, but shouldn't it be possible to define `conn_cbs` without the guard, and then guard all calls to it with `IS_ENABLED(CONFIG_BT_CONN_DYNAMIC_CALLBACKS)`, and then have the linker remove `conn_cbs` from memory?